CritBit/src/org/zoodb/index/critbit/CritBit64COW.java

```diff
@@ -8,21 +8,14 @@
 import java.util.Iterator;
 import java.util.NoSuchElementException;
 import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantLock;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class CritBit64COW<V> implements Iterable<V> {
 
     private final int DEPTH = 64;
     private Lock writeLock = new ReentrantLock();
 
-    private Node<V> root;
-    //the following contains either the actual key or the prefix for the sub-node
-    private long rootKey;
-    private V rootVal;
-
-    private int size;
+    private AtomicInfo<V> info = new AtomicInfo<>();
```

Comment:
Maybe this should be 'final' to avoid accidental assignment (unless I misunderstand the code...)